packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -113,6 +117,18 @@ class HotRunner extends ResidentRunner {
     for (FlutterDevice device in flutterDevices)
       device.initLogReader();
 
+    if (previewDart2 && (generator == null)) {
+      generator = new ResidentCompiler(
+        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath));
+      final String outputFile = await generator.compile(mainPath);
+      // If compilation was successful, no reason VM should reject it.
```

Comment:
Aren't there program changes that the VM will reject? for example changing a class into an enum?

It seems to me that we want to delay calling `accept` until after the reload is label a success.